### ISSUE TITLE
Design token additional properties (description, type, extensions) need to be in 5.2.x instead of 5.3, 5.4 etc.

### ISSUE THREAD

**Issue by lauthieb** [design-tokens/community-group#154]
Hello,

_First of all, a huge thank you to all the people who have contributed so far to this wonderful initiative.
Being passionate about UI and standardization, I loved reading the second draft and I want to be part of the contributors by proposing some changes that I hope you will like._

## Context

For groups (section 6), you did this:

<img width="296" alt="CleanShot 2022-07-01 at 02 30 08@2x" src="https://user-images.githubusercontent.com/9600228/176799706-f62d106b-ebe5-4a48-886e-e34a2347dde7.png">

## Proposal

I think it will be clearer to do the same for the section 5.2 which is very small:

<img width="876" alt="CleanShot 2022-07-01 at 02 28 12@2x" src="https://user-images.githubusercontent.com/9600228/176799521-cf33ddaa-5eef-4c75-8eaa-4190c5c0a93a.png">

Instead of using 5.3, 5.4, 5.5 for description, type & extensions respectively. It could be better to apply them inside section 5.2.x.

What do you think about this?

Thanks in advance for your opinion!

**Comment by c1rrus** [comment]
Agreed, that would be more consistent.

However, I've recently been wondering if we should restructure our outline more dramatically than that:

The `$description` exists for both groups and tokens, _and_ it has the same purpose in both cases (i.e. it provides a description of the group or token it's added to). Currently, `$extensions` only exists on tokens, but there's a proposal (which I agree with) to add it to groups too (#106).

The `$type` property also exists for both groups and tokens. Although it's meaning is slightly different in each case (On tokens it sets that token's type. On groups it sets a default type that is inherited by any child tokens which do not have their own `$type` property), everything else about it is the same in both cases.

Several new properties are being proposed (e.g. `$deprecated`, `$usedFor`, etc.) and I think many of those will be applicable to groups and tokens too.

If our spec keeps the same heading structure, we're likely end up with a lot of duplication.

I've therefore been wondering whether we should reorganise our outline into something like this:

1. Entities
    1. Design tokens
        * Explain the purpose and syntax of design tokens
    1. Groups
        * Explain the purpose and syntax of groups
1. Properties
    1. `$value`
        * Explanation of which entities this prop can be applied to (groups only, tokens only or both), it's purpose, permitted values, etc. 
    3. `$type`
    4. `$description`
    5. `$extensions`
    6. ...

While I'm pretty sure that would make the spec easier to write and maintain, the more important factor (IMHO) is whether that would make it easier to read and understand. Keen to hear what you all think!

**Comment by lauthieb** [comment]
Hi @c1rrus, oh yes you're entirely right, a lot of different sub-sections are used for tokens & groups so we can mix them together. Agree with your proposal. 
Can't wait to see other people agree or challenge this :)
Thanks for your help!

**Comment by kevinmpowell** [comment]
@c1rrus I agree the current structure creates duplication of headers, but I find that structure more scannable when looking at the outline. Your proposed changes will DRY up the document, but I don't think that's worth the tradeoff of losing a more scannable outline.

Combining token and group definitions for each property will increase the complexity of each of those explanations and diminish the readability of those definitions IMO.

I propose we keep **Design Token** and **Groups** as top-level sections and adjust the heading levels beneath to create alignment as originally proposed by @lauthieb.
